### PR TITLE
Trackers: frequency is a cron string

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Expect that the field frequency is a cron string over an enum. 
Change the way we fetch tracker to notify based on this cron string and the last time they were processed. 

## Risk

Can be rolled back. 

## Deploy Plan

Deploy front;

### CODE REVIEW

File: front/lib/resources/tracker_resource.ts [hunk 1]
```diff
@@ -312,19 +314,26 @@ export class TrackerConfigurationResource extends ResourceWithSpace<TrackerConfi
...
-            consumedAt: null,
-          },
-        },
       ],
     });
 
-    const filteredTrackers = trackers.filter(
-      (tracker) => tracker.generations.length
-    );
+    const filteredTrackers = trackers.filter((tracker) => {
```
Comment (PopDaph) [inline front/lib/resources/tracker_resource.ts:320]: Ideally we have some basic tests to validate this logic. 😖

**Review by fontanierh — APPROVED** [review]
naice

File: front/lib/resources/tracker_resource.ts [hunk 2]
```diff
@@ -300,33 +302,60 @@ export class TrackerConfigurationResource extends ResourceWithSpace<TrackerConfi
...
+  // Fetches the active trackers that need to be processed for notifications.
+  static async internalFetchTrackersToNotify(
+    currentRunMs: number
+  ): Promise<TrackerIdWorkspaceId[]> {
+    if (!currentRunMs || typeof currentRunMs !== "number") {
+      throw new Error("Invalid currentRunMs parameter");
+    }
+
+    // Look back 20 minutes to ensure we don't miss any runs.
+    const LOOK_BACK_PERIOD_MS = 1 * 20 * 60 * 1000; // 20 minutes.
```
Comment (fontanierh) [inline front/lib/resources/tracker_resource.ts:314]: Why is this needed ?

File: front/lib/resources/tracker_resource.ts [hunk 3]
```diff
@@ -300,33 +302,60 @@ export class TrackerConfigurationResource extends ResourceWithSpace<TrackerConfi
...
+
+      try {
+        const interval = parseExpression(tracker.frequency, {
+          currentDate: tracker.lastNotifiedAt ?? tracker.createdAt, // Start from the last run to avoid missing a run.
+        });
+        const nextExpectedRunMs = interval.next().getTime();
+
+        return nextExpectedRunMs <= lookBackMs;
+      } catch (e) {
+        logger.error(
```
Comment (fontanierh) [inline front/lib/resources/tracker_resource.ts:348]: shouldn't this just fail the whole thing ?

**Review by fontanierh — APPROVED** [review]
LGTM
